h/session.py

```diff
@@ -11,13 +11,27 @@ def model(request):
     session['userid'] = request.authenticated_userid
     session['groups'] = _current_groups(request)
     session['features'] = request.feature.all()
-    session['preferences'] = {}
-    user = request.authenticated_user
-    if user and not user.sidebar_tutorial_dismissed:
-        session['preferences']['show_sidebar_tutorial'] = True
+    session['preferences'] = _user_preferences(request.authenticated_user)
     return session
 
 
+def profile(request):
+    """
+    Return a representation of the current user's information and settings.
+
+    This is intended for use in a third-party setting, where the client will
```

Comment:
"for use in a third-party setting" is a little ambiguous. Initially we will be using it in the client when using third-party _accounts_, but I expect in the not-too-distant future we will use OAuth for first-party accounts as well. This API is also likely to be useful to third-party _clients_.